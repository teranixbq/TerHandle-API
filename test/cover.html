
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>service: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">terhandle/internal/features/admin/service/service.go (76.0%)</option>
				
				<option value="file1">terhandle/internal/features/feedback/service/service.go (89.5%)</option>
				
				<option value="file2">terhandle/internal/features/request-teknisi/service/service.go (82.9%)</option>
				
				<option value="file3">terhandle/internal/features/user/service/service.go (80.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package service

import (
        "terhandle/internal/features/admin/entity"
)

type AdminService struct {
        AdminRepository entity.AdminRepositoryInterface
}

func NewUserService(Ad entity.AdminRepositoryInterface) entity.AdminServiceInterface <span class="cov8" title="1">{
        return &amp;AdminService{
                AdminRepository: Ad,
        }
}</span>

func (Ad *AdminService) Create(payload entity.CoreProfesi) error <span class="cov8" title="1">{

        err := Ad.AdminRepository.Insert(payload)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (Ad *AdminService) GetAll() ([]entity.CoreProfesi, error) <span class="cov8" title="1">{
        profesiAll, err := Ad.AdminRepository.SelectAll()
        if err != nil </span><span class="cov0" title="0">{
                return nil, err
        }</span>

        <span class="cov8" title="1">return profesiAll, err</span>
}

func (Ad *AdminService) Update(id_profesi uint, payload entity.CoreProfesi) error <span class="cov8" title="1">{
        err := Ad.AdminRepository.Update(id_profesi, payload)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (Ad *AdminService) Delete(id_profesi uint) error <span class="cov8" title="1">{
        err := Ad.AdminRepository.Delete(id_profesi)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (Ad *AdminService) InsertBiaya(payload entity.CoreTransport) error <span class="cov8" title="1">{
        err := Ad.AdminRepository.InsertBiaya(payload)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return err</span>
}

func (Ad *AdminService) UpdateBiaya(id uint, payload entity.CoreTransport) error <span class="cov8" title="1">{
        err := Ad.AdminRepository.UpdateBiaya(id, payload)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return err</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package service

import (
        "errors"
        "terhandle/internal/features/feedback/entity"
)

type feedbackService struct {
        feedbackRepository entity.FeedbackRepositoryInterface
}

func NewUserService(fr entity.FeedbackRepositoryInterface) entity.FeedbackServiceInterface <span class="cov8" title="1">{
        return &amp;feedbackService{
                feedbackRepository: fr,
        }
}</span>

func (fr *feedbackService) Create(payload entity.CoreFeedback) error <span class="cov8" title="1">{

        if payload.TeknisiID == 0 || payload.UsersID == 0 || payload.RequestTeknisiID == 0 || payload.Ulasan == "" </span><span class="cov8" title="1">{
                return errors.New("harap lengkapi data dengan benar")
        }</span>

        <span class="cov8" title="1">if payload.Rating &gt; 5 </span><span class="cov8" title="1">{
                return errors.New("tidak boleh lebih dari lima")
        }</span>
        
        <span class="cov8" title="1">err := fr.feedbackRepository.SelectByIdAndRole(payload.UsersID, payload.TeknisiID, "user", "teknisi")
        if err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">err = fr.feedbackRepository.SelectByIdAndIdUser(payload.RequestTeknisiID, payload.UsersID, payload.TeknisiID)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">err = fr.feedbackRepository.Insert(payload.RequestTeknisiID,payload.TeknisiID,payload)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (fr *feedbackService) UpdateFeedback(userid ,id_feedback,id_request uint, payload entity.CoreFeedback) error<span class="cov8" title="1">{

        err := fr.feedbackRepository.UpdateFeedback(userid,id_feedback,id_request,payload)
        if err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		<pre class="file" id="file2" style="display: none">package service

import (
        "errors"
        "terhandle/internal/features/request-teknisi/entity"
)

type userService struct {
        userRepository entity.UserRequestRepositoryInterface
}

func NewUserService(ur entity.UserRequestRepositoryInterface) entity.UserRequestServiceInterface <span class="cov8" title="1">{

        return &amp;userService{userRepository: ur}
}</span>

func (us *userService) Create(payload entity.Core) error <span class="cov8" title="1">{

        if payload.TeknisiID == 0 || payload.UsersID == 0 || payload.Deskripsi == ""</span><span class="cov8" title="1">{
                return errors.New("harap lengkapi dengan benar")
        }</span>
        <span class="cov8" title="1">err := us.userRepository.SelectByIdAndRole(int(payload.UsersID), int(payload.TeknisiID), "user", "teknisi")
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">err = us.userRepository.Insert(payload)
        if err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) CreateUpdateDetail(userid int, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.Update(userid, data); err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) GetAllById(userid int) ([]entity.Core, error) <span class="cov8" title="1">{
        history, err := us.userRepository.SelectAllById(userid)
        if err != nil </span><span class="cov0" title="0">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return history, nil</span>
}

func (us *userService) GetById(userid, id int) ([]entity.Core, error) <span class="cov8" title="1">{
        result, err := us.userRepository.SelectById(userid, id)
        if err != nil </span><span class="cov0" title="0">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return result, nil</span>
}

func (us *userService) ClaimRequest(id_request int, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.UpdateClaims(id_request, data); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">if err := us.userRepository.UpdateField(id_request, "status", "konfirmasi biaya"); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) KonfirmasiBiaya(id_user, id_request int, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.UpdateStatusKonfirmBiaya(id_user, id_request, data); err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">if err := us.userRepository.UpdateField(id_request, "status", "diproses"); err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) BatalkanRequest(id_user, id_request int, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.UpdateStatusBatal(id_user, id_request, data); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>
        <span class="cov8" title="1">if err := us.userRepository.UpdateField(id_request, "status", "dibatalkan"); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) SelesaikanRequest(id_user, id_request int, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.UpdateStatusSelesai(id_user, id_request, data); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>
        <span class="cov8" title="1">if err := us.userRepository.UpdateField(id_request, "status", "selesai"); err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		<pre class="file" id="file3" style="display: none">package service

import (
        "errors"
        "fmt"
        "terhandle/internal/features/user/dto"
        "terhandle/internal/features/user/entity"

        "github.com/go-playground/validator/v10"
        "github.com/go-sql-driver/mysql"
)

type userService struct {
        userRepository entity.UserRepositoryInterface
        validate       *validator.Validate
}

func NewUserService(ur entity.UserRepositoryInterface) entity.UserServiceInterface <span class="cov8" title="1">{

        return &amp;userService{userRepository: ur, validate: validator.New()}
}</span>

func (us *userService) Create(payload entity.Core) error <span class="cov8" title="1">{
        fmt.Println(payload.Email,payload.Password,payload.Nama)
        if payload.Email == "" || payload.Password == "" || payload.Nama == "" </span><span class="cov0" title="0">{
                return errors.New("error. email dan password harus diisi")
        }</span>

        // errValidate := us.validate.Struct(payload)

        // if errValidate != nil {
        //         return errValidate
        // }

        <span class="cov8" title="1">err := us.userRepository.Insert(payload)
        if err != nil </span><span class="cov8" title="1">{
                if mysqlErr, ok := err.(*mysql.MySQLError); ok </span><span class="cov0" title="0">{
                        if mysqlErr.Number == 1062 </span><span class="cov0" title="0">{
                                return errors.New("email sudah terdaftar. Silakan gunakan email lain")
                        }</span>
                }
                <span class="cov8" title="1">return err</span>
        }
        <span class="cov8" title="1">return nil</span>
}

func (us *userService) Login(email, password string) (entity.Core, string, error) <span class="cov8" title="1">{
        if email == "" || password == ""</span><span class="cov8" title="1">{
                return entity.Core{}, "",errors.New("email dan password harus diisi")
        }</span>

        <span class="cov8" title="1">user := dto.RequestLogin{
                Email:    email,
                Password: password,
        }

        data, token, err := us.userRepository.Login(user.Email, user.Password)
        if err != nil </span><span class="cov8" title="1">{
                return data, "", err
        }</span>
        <span class="cov0" title="0">return data, token, nil</span>
}

func (us *userService) CreateUpdateDetail(id_user uint, data entity.Core) error <span class="cov8" title="1">{

        if err := us.userRepository.Update(id_user, data); err != nil </span><span class="cov8" title="1">{
                return err
        }</span>

        <span class="cov0" title="0">return nil</span>
}

func (us *userService) RequestTeknisiRole(id_user uint) error <span class="cov8" title="1">{

        if err := us.userRepository.UpdateField(id_user, "role", "teknisi"); err != nil </span><span class="cov0" title="0">{
                return err
        }</span>

        <span class="cov8" title="1">return nil</span>
}

func (us *userService) GetAll() ([]entity.Core, error) <span class="cov8" title="1">{
        users, err := us.userRepository.SelectAll()
        if err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return users, err</span>
}

func (us *userService) GetUserById(id_user uint, role string) (entity.Core, error) <span class="cov8" title="1">{
        users, err := us.userRepository.SelectUserById(id_user,role)
        if err != nil </span><span class="cov8" title="1">{
                return entity.Core{}, err
        }</span>

        <span class="cov0" title="0">return users, err</span>
}

func (us *userService) GetById(id_user uint) ([]entity.Core, error) <span class="cov8" title="1">{
        users, err := us.userRepository.SelectByIdWithFeedback(id_user)
        if err != nil </span><span class="cov8" title="1">{
                return nil, err
        }</span>
        <span class="cov8" title="1">return users, err</span>
}


func (us *userService) DeleteById(id_user uint) error <span class="cov8" title="1">{
        err := us.userRepository.DeleteById(id_user)
        if err != nil </span><span class="cov0" title="0">{
                return nil
        }</span>
        <span class="cov8" title="1">return nil</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
