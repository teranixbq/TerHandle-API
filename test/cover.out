mode: set
terhandle/internal/features/admin/service/service.go:11.86,15.2 1 1
terhandle/internal/features/admin/service/service.go:17.66,20.16 2 1
terhandle/internal/features/admin/service/service.go:20.16,22.3 1 0
terhandle/internal/features/admin/service/service.go:24.2,24.12 1 1
terhandle/internal/features/admin/service/service.go:27.64,29.16 2 1
terhandle/internal/features/admin/service/service.go:29.16,31.3 1 0
terhandle/internal/features/admin/service/service.go:33.2,33.24 1 1
terhandle/internal/features/admin/service/service.go:36.83,38.16 2 1
terhandle/internal/features/admin/service/service.go:38.16,40.3 1 0
terhandle/internal/features/admin/service/service.go:42.2,42.12 1 1
terhandle/internal/features/admin/service/service.go:45.55,47.16 2 1
terhandle/internal/features/admin/service/service.go:47.16,49.3 1 0
terhandle/internal/features/admin/service/service.go:51.2,51.12 1 1
terhandle/internal/features/admin/service/service.go:54.73,56.16 2 1
terhandle/internal/features/admin/service/service.go:56.16,58.3 1 0
terhandle/internal/features/admin/service/service.go:60.2,60.12 1 1
terhandle/internal/features/admin/service/service.go:63.82,65.16 2 1
terhandle/internal/features/admin/service/service.go:65.16,67.3 1 0
terhandle/internal/features/admin/service/service.go:69.2,69.12 1 1
terhandle/internal/features/feedback/service/service.go:12.92,16.2 1 1
terhandle/internal/features/feedback/service/service.go:18.70,20.109 1 1
terhandle/internal/features/feedback/service/service.go:20.109,22.3 1 1
terhandle/internal/features/feedback/service/service.go:24.2,24.24 1 1
terhandle/internal/features/feedback/service/service.go:24.24,26.3 1 1
terhandle/internal/features/feedback/service/service.go:28.2,29.16 2 1
terhandle/internal/features/feedback/service/service.go:29.16,31.3 1 1
terhandle/internal/features/feedback/service/service.go:33.2,34.16 2 1
terhandle/internal/features/feedback/service/service.go:34.16,36.3 1 0
terhandle/internal/features/feedback/service/service.go:38.2,39.16 2 1
terhandle/internal/features/feedback/service/service.go:39.16,41.3 1 0
terhandle/internal/features/feedback/service/service.go:43.2,43.12 1 1
terhandle/internal/features/feedback/service/service.go:46.114,49.16 2 1
terhandle/internal/features/feedback/service/service.go:49.16,51.3 1 1
terhandle/internal/features/feedback/service/service.go:53.2,53.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:12.98,15.2 1 1
terhandle/internal/features/request-teknisi/service/service.go:17.58,19.78 1 1
terhandle/internal/features/request-teknisi/service/service.go:19.78,21.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:22.2,23.16 2 1
terhandle/internal/features/request-teknisi/service/service.go:23.16,25.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:27.2,28.16 2 1
terhandle/internal/features/request-teknisi/service/service.go:28.16,30.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:32.2,32.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:35.79,37.63 1 1
terhandle/internal/features/request-teknisi/service/service.go:37.63,39.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:41.2,41.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:44.70,46.16 2 1
terhandle/internal/features/request-teknisi/service/service.go:46.16,48.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:49.2,49.21 1 1
terhandle/internal/features/request-teknisi/service/service.go:52.71,54.16 2 1
terhandle/internal/features/request-teknisi/service/service.go:54.16,56.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:57.2,57.20 1 1
terhandle/internal/features/request-teknisi/service/service.go:60.77,62.73 1 1
terhandle/internal/features/request-teknisi/service/service.go:62.73,64.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:66.2,66.96 1 1
terhandle/internal/features/request-teknisi/service/service.go:66.96,68.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:70.2,70.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:73.89,75.94 1 1
terhandle/internal/features/request-teknisi/service/service.go:75.94,77.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:79.2,79.88 1 1
terhandle/internal/features/request-teknisi/service/service.go:79.88,81.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:83.2,83.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:86.89,88.87 1 1
terhandle/internal/features/request-teknisi/service/service.go:88.87,90.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:91.2,91.90 1 1
terhandle/internal/features/request-teknisi/service/service.go:91.90,93.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:95.2,95.12 1 1
terhandle/internal/features/request-teknisi/service/service.go:98.91,100.89 1 1
terhandle/internal/features/request-teknisi/service/service.go:100.89,102.3 1 1
terhandle/internal/features/request-teknisi/service/service.go:103.2,103.87 1 1
terhandle/internal/features/request-teknisi/service/service.go:103.87,105.3 1 0
terhandle/internal/features/request-teknisi/service/service.go:107.2,107.12 1 1
terhandle/internal/features/user/service/service.go:18.84,21.2 1 1
terhandle/internal/features/user/service/service.go:23.58,25.73 2 1
terhandle/internal/features/user/service/service.go:25.73,27.3 1 0
terhandle/internal/features/user/service/service.go:35.2,36.16 2 1
terhandle/internal/features/user/service/service.go:36.16,37.50 1 1
terhandle/internal/features/user/service/service.go:37.50,38.31 1 0
terhandle/internal/features/user/service/service.go:38.31,40.5 1 0
terhandle/internal/features/user/service/service.go:42.3,42.13 1 1
terhandle/internal/features/user/service/service.go:44.2,44.12 1 1
terhandle/internal/features/user/service/service.go:47.83,48.34 1 1
terhandle/internal/features/user/service/service.go:48.34,50.3 1 1
terhandle/internal/features/user/service/service.go:52.2,58.16 3 1
terhandle/internal/features/user/service/service.go:58.16,60.3 1 1
terhandle/internal/features/user/service/service.go:61.2,61.25 1 0
terhandle/internal/features/user/service/service.go:64.81,66.64 1 1
terhandle/internal/features/user/service/service.go:66.64,68.3 1 1
terhandle/internal/features/user/service/service.go:70.2,70.12 1 0
terhandle/internal/features/user/service/service.go:73.63,75.82 1 1
terhandle/internal/features/user/service/service.go:75.82,77.3 1 0
terhandle/internal/features/user/service/service.go:79.2,79.12 1 1
terhandle/internal/features/user/service/service.go:82.56,84.16 2 1
terhandle/internal/features/user/service/service.go:84.16,86.3 1 1
terhandle/internal/features/user/service/service.go:87.2,87.19 1 1
terhandle/internal/features/user/service/service.go:90.84,92.16 2 1
terhandle/internal/features/user/service/service.go:92.16,94.3 1 1
terhandle/internal/features/user/service/service.go:96.2,96.19 1 0
terhandle/internal/features/user/service/service.go:99.69,101.16 2 1
terhandle/internal/features/user/service/service.go:101.16,103.3 1 1
terhandle/internal/features/user/service/service.go:104.2,104.19 1 1
terhandle/internal/features/user/service/service.go:108.55,110.16 2 1
terhandle/internal/features/user/service/service.go:110.16,112.3 1 0
terhandle/internal/features/user/service/service.go:113.2,113.12 1 1
